apps/src/lib/node/ledger/shell/governance.rs

```diff
@@ -340,64 +347,71 @@ where
     Ok(true)
 }
 
-fn execute_pgf_payment_proposal<S>(
-    storage: &mut S,
+fn execute_pgf_payment_proposal<D, H>(
+    storage: &mut WlStorage<D, H>,
     token: &Address,
     payments: Vec<PGFAction>,
     proposal_id: u64,
 ) -> Result<bool>
 where
-    S: StorageRead + StorageWrite,
+    D: DB + for<'iter> DBIter<'iter> + Sync + 'static,
+    H: StorageHasher + Sync + 'static,
 {
     for payment in payments {
         match payment {
             PGFAction::Continuous(action) => match action {
                 AddRemove::Add(target) => {
                     pgf_storage::fundings_handle().insert(
                         storage,
-                        target.target.clone(),
+                        target.target().clone(),
                         StoragePgfFunding::new(target.clone(), proposal_id),
                     )?;
                     tracing::info!(
                         "Execute ContinousPgf from proposal id {}: set {} to \
                          {}.",
                         proposal_id,
-                        target.amount.to_string_native(),
-                        target.target
+                        target.amount().to_string_native(),
+                        target.target()
                     );
                 }
                 AddRemove::Remove(target) => {
                     pgf_storage::fundings_handle()
-                        .remove(storage, &target.target)?;
+                        .remove(storage, &target.target())?;
                     tracing::info!(
                         "Execute ContinousPgf from proposal id {}: set {} to \
                          {}.",
                         proposal_id,
-                        target.amount.to_string_native(),
-                        target.target
+                        target.amount().to_string_native(),
+                        target.target()
                     );
                 }
             },
             PGFAction::Retro(target) => {
-                match token::transfer(
-                    storage,
-                    token,
-                    &ADDRESS,
-                    &target.target,
-                    target.amount,
-                ) {
+                let result = match &target {
+                    PGFTarget::Internal(target) => token::transfer(
+                        storage,
+                        token,
+                        &ADDRESS,
+                        &target.target,
+                        target.amount,
+                    ),
+                    PGFTarget::Ibc(target) => {
```

Comment:
This actually adds support for rPGF payments over ibc which doesn't show up as a requirement in #1395. It looks good to me I just want to make sure this is intended

Reply:
ah I think having both is good